### PR TITLE
feat(inputs.radius): Provide setting to set request IP address

### PR DESCRIPTION
**Pull request by powersj**
## Checklist

- [x] No AI generated code was used in this PR

## Related issues

fixes: #14962

### CODE REVIEW

**Comment by srebhan** [comment]
@powersj according to [RFC2865](https://datatracker.ietf.org/doc/html/rfc2865#page-29) the field (or NAS-Identifier) **MUST** be present for the access-request packet we are sending. So I think we should add a `nas_identifier` option and use this _or_ determine the `LocalAddr` of the client and use it as `NAS-IP-Address`...

**Comment by powersj** [comment]
PR updated with the default set to 127.0.0.1 if empty.

File: plugins/inputs/radius/sample.conf [hunk 1]
```diff
@@ -7,5 +7,8 @@
   password = "mypassword"
   secret = "mysecret"
 
+  ## Request source server IP, normally the server running telegraf.
```
Comment (srebhan) [inline plugins/inputs/radius/sample.conf:10]:
```suggestion
  ## Request source server IP, normally the server running telegraf.
  ## This corresponds to Radius' NAS-IP-Address.
```

**Review by srebhan — COMMENTED** [review]
Looks good. My only comment would be to reference the term NAS-IP-Address or we name the option like this (as you had before :-()... Not sure which approach is better....

**Review by srebhan — APPROVED** [review]

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**